src/NHibernate/Impl/StatelessSessionImpl.cs

```diff
@@ -228,7 +228,20 @@ public override void AfterTransactionCompletion(bool successful, ITransaction tx
 
 		public override object GetContextEntityIdentifier(object obj)
 		{
-			return null;
+			using (BeginProcess())
+			{
+				if (obj.IsProxy())
```

Comment:
I gave it another thought. I think the `true` branch is wrong/not possible/not needed.